control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -665,25 +670,36 @@ Path AEB::generateEgoPath(const double curr_v, const double curr_w)
     return path;
   }
 
-  const double & horizon = imu_prediction_time_horizon_;
+  const double horizon = imu_prediction_time_horizon_;
+  const double base_link_to_front_offset = vehicle_info_.max_longitudinal_offset_m;
+  const double rear_overhang = vehicle_info_.rear_overhang_m;
+  const double vehicle_half_width = expand_width_ + vehicle_info_.vehicle_width_m / 2.0;
+
   double path_arc_length = 0.0;
   double t = 0.0;
-
   bool finished_creating_path = false;
+  const auto longitudinal_offset = (curr_v > 0.0) ? base_link_to_front_offset : -rear_overhang;
+  const auto lateral_offset = (curr_v * curr_w > 0.0) ? vehicle_half_width : -vehicle_half_width;
+
```

Comment:
@danielsanchezaran 
```suggestion
  double curr_x = 0.0;
  double curr_y = 0.0;
  double curr_yaw = 0.0;
  double path_arc_length = 0.0;
  double t = 0.0;
  while (true) {
    curr_x += curr_v * std::cos(curr_yaw) * dt;
    curr_y += curr_v * std::sin(curr_yaw) * dt;
    curr_yaw += curr_w * dt;

    geometry_msgs::msg::Pose current_pose =
      autoware::universe_utils::calcOffsetPose(ini_pose, curr_x, curr_y, 0.0, curr_yaw);

    const auto edge_of_ego_vehicle = autoware::universe_utils::calcOffsetPose(
                                       current_pose, longitudinal_offset, lateral_offset, 0.0)
                                       .position;

    t += dt;
    path_arc_length += distance_between_points;
    const bool path_length_threshold_surpassed = path_arc_length > max_generated_imu_path_length_;
    const bool lat_dev_threshold_surpassed =
      limit_imu_path_lat_dev_ && std::abs(edge_of_ego_vehicle.y) > imu_path_lat_dev_threshold_;
    const bool basic_path_conditions_satisfied =
      (t > horizon) && (path_arc_length > min_generated_imu_path_length_);

    if (
      basic_path_conditions_satisfied || path_length_threshold_surpassed ||
      lat_dev_threshold_surpassed) {
      break;
    }
    path.push_back(current_pose);
  }
```
propose some minor changes (including my personal preferences) to the code, while keeping the same processing flow. While the current implementation is acceptable, please consider incorporating these improvements if you find them beneficial. Note that I've also slightly modified the variable declaration positions, so directly applying these changes won't compile successfully.

Reply:
Thank you, I will add this suggestions

Reply:
@kyoichi-sugahara done, could you check again please?

Reply:
@danielsanchezaran 
Thank you LGTM :+1: